.github/workflows/branch-build-test.yml

```diff
@@ -0,0 +1,41 @@
+name: branch - Build & Test
```

Comment:
What's the advantage of having a different build configuration for branches versus PRs and master?

Reply:
They seem the same for now, so it might be best to just coalesce them together.

Reply:
They're mostly the same.  It replicates ICU process (all pushes to master and all PRs get run on all 3 OSes), but with the added functionality that non-master branches get run on Linux only on every push.  That extra bit would take effect in people's personal forks when they're pushing a branch.

I thought it would be an extra level of testing so people can have a chance to fix up branches with failing integration tests in their personal forks before initiating a PR and finding out later.  But I can easily consolidate them if that makes more sense.

Reply:
I went ahead and de-duped the workflows to have just one.  We can easily go back to something more custom in the future, just lmk.